docs/api.md

```diff
@@ -0,0 +1,503 @@
+
+Response:
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector" | "streams",
+    "result": [<vector value>] | [<stream value>]
+  }
+}
+```
+
+Where `<vector value>` is:
+
+```
+{
+  "metric": {
+    <label key-value pairs>
+  },
+  "value": [
+    <number: nanosecond unix epoch>,
+    <string: value>
+  ]
+}
+```
+
+And `<stream value>` is:
+
+```
+{
+  "labels": "<LogQL label key-value pairs>",
+  "entries": [
+    {
+      "ts": "<RFC3339Nano string>",
+      "line": "<log line>",
+    },
+    ...
+  ]
+}
+```
+
+### Examples
+
+```bash
+$ curl -G -s  "http://localhost:3100/loki/api/v1/query" --data-urlencode 'query=sum(rate({job="varlogs"}[10m])) by (level)' | jq
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {},
+        "value": [
+          1559848867745737,
+          "1267.1266666666666"
+        ]
+      },
+      {
+        "metric": {
+          "level": "warn"
+        },
+        "value": [
+          1559848867745737,
+          "37.77166666666667"
+        ]
+      },
+      {
+        "metric": {
+          "level": "info"
+        },
+        "value": [
+          1559848867745737,
+          "37.69"
+        ]
+      }
+    ]
+  }
+}
+```
+
+```bash
+$ curl -G -s  "http://localhost:3100/loki/api/v1/query" --data-urlencode 'query={job="varlogs"}' | jq
+{
+  "status": "success",
+  "data": {
+    "resultType": "streams",
+    "result": [
+      {
+        "labels": "{filename=\"/var/log/myproject.log\", job=\"varlogs\", level=\"info\"}",
+        "entries": [
+          {
+            "ts": "2019-06-06T19:25:41.972739Z",
+            "line": "foo"
+          },
+          {
+            "ts": "2019-06-06T19:25:41.972722Z",
+            "line": "bar"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+## `GET /loki/api/v1/query_range`
+
+`/loki/api/v1/query_range` is used to do a query over a range of time and
+accepts the following query parameters in the URL:
+
+- `query`: The [LogQL](./logql.md) query to perform
+- `limit`: The max number of entries to return
+- `start`: The start time for the query as a nanosecond Unix epoch. Defaults to one hour ago.
+- `end`: The start time for the query as a nanosecond Unix epoch. Defaults to now.
+- `step`: Query resolution step width in seconds. Defaults to 1.
+- `direction`: Determines the sort order of logs. Supported values are `forward` or `backward`. Defaults to `backward.`
+
+Requests against this endpoint require Loki to query the index store in order to
+find log streams for particular labels. Because the index store is spread out by
+time, the time span covered by `start` and `end`, if large, may cause additional
+load against the index server and result in a slow query.
+
+Response:
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "matrix" | "streams",
```

Comment:
we should reference to Prometheus documentation for `vector` and `matrix` and add a section to describe `streams` may be ?